splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/index/IndexTransformer.java

```diff
@@ -707,18 +707,12 @@ public int getMaxBaseColumnPosition() {
     }
 
     private LanguageConnectionContext getLcc(DDLMessage.TentativeIndex tentativeIndex) throws StandardException {
-        boolean prepared = false;
-        SpliceTransactionResourceImpl transactionResource = null;
-        try {
-            TxnView txn = DDLUtils.getLazyTransaction(tentativeIndex.getTxnId());
-            transactionResource = new SpliceTransactionResourceImpl();
-            prepared = transactionResource.marshallTransaction(txn);
+        TxnView txn = DDLUtils.getLazyTransaction(tentativeIndex.getTxnId());
```

Comment:
Returning the LCC here seems wrong. Rework that class a bit to not have to return a LCC whose contextmanager got closed

Reply:
lcc is only used in getExecutableIndexExpression(). You can refactor to marshall transactions there only when it is being used

Reply:
There is a dedicated jira for this specific issue: DB-10485. Zhen is currently working on it, so I would prefer not modify this further.